### PR TITLE
Add test.yml to run test on Github CI

### LINKED ISSUE #5: Run tests on CI
The goal is to make sure tests pass when code is pushed or when PRs are opened and merged. Make sure to run mix compile with `mix compile --warnings-as-errors` to make sure there are no compilation warnings.

### PR DESCRIPTION
**Pull request by Atul9**
Closes #5    
CI output: [here](https://github.com/Atul9/fika/runs/1194068572?check_suite_focus=true)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+on: push
```
Comment (emilsoman) [inline .github/workflows/test.yml]: Shouldn't this be `on: [pull_request, push]`

Comment (Atul9) [reply .github/workflows/test.yml]: yes, I have updated the PR.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-elixir@v1
+        with:
+          otp-version: '22.2'
+          elixir-version: '1.10.4'
+      - run: mix deps.get
+      - run: mix compile --warnings-as-errors
```
Comment (polvalente) [inline .github/workflows/test.yml:13]: I reckon it would be nice for these steps to be cached

Comment (polvalente) [reply .github/workflows/test.yml:13]: Also, I would recommend adding a job for Credo checks and `mix format --check-formatted`, to ensure code uniformity

**Comment by emilsoman** [comment]
Thanks @Atul9 👏

**Pull request merged**